src/components/Editor/index.js

```diff
@@ -436,14 +427,17 @@ class Editor extends PureComponent<Props, State> {
     const { selectedLocation, selectedSource } = nextProps;
 
     if (selectedLocation && this.shouldScrollToLocation(nextProps)) {
-      let { line, column } = toEditorPosition(selectedLocation);
+      const line = selectedLocation.line;
+      let column = selectedLocation.column;
+      // let { line, column } = toEditorPosition(nextProps.selectedLocation);
 
       if (selectedSource && hasDocument(selectedSource.id)) {
         const doc = getDocument(selectedSource.id);
-        const lineText: ?string = doc.getLine(line);
+        const lineText = doc.getLineContent(line);
         column = Math.max(column, getIndentation(lineText));
       }
-      scrollToColumn(editor.codeMirror, line, column);
+
+      editor.monaco.revealPosition({ lineNumber: line, column: column });
```

Comment:
one of our concerns is we don't want to scroll every time the user moves, just when it is not visible. Does this do that?

Reply:
The position is revealed only when necessary, which means it only scrolls when it's not visible. We can also use `revealPositionInCenter` if we want to reveal the position to the center of the viewport when necessary.